### PR TITLE
Add Craft, Smelt and Pickup events.

### PR DESCRIPTION
**Pull request by gegy1000**

### CODE REVIEW

File: src/main/java/xyz/openmodloader/event/impl/EntityEvent.java [hunk 1]
```diff
@@ -244,4 +245,40 @@ public EntityLightningBolt getLightningBolt() {
...
+            super(entity);
+            this.item = item;
+        }
+
+        /**
+         * Gets the item being picked up.
+         *
+         * @return the item being picked up.
+         */
+        public EntityItem getItem() {
```
Comment (iLexiconn) [inline src/main/java/xyz/openmodloader/event/impl/EntityEvent.java]: maybe add a setter?

File: src/main/java/xyz/openmodloader/event/impl/PlayerEvent.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+         */
+        public ItemStack getResult() {
+            return result;
+        }
+    }
+
+    /**
+     * This event is fired when a player picks a smelted item out of a furnace.
+     */
+    public static class Smelt extends PlayerEvent {
```
Comment (iLexiconn) [inline src/main/java/xyz/openmodloader/event/impl/PlayerEvent.java]: place this in `BlockEntity`

File: src/main/java/xyz/openmodloader/event/impl/PlayerEvent.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+     * @return the player that fired this event.
+     */
+    public EntityPlayer getPlayer() {
+        return player;
+    }
+
+    /**
+     * This event is fired when a player crafts an item with a crafting table.
+     */
+    public static class Craft extends PlayerEvent {
```
Comment (iLexiconn) [inline src/main/java/xyz/openmodloader/event/impl/PlayerEvent.java]: place this in `BlockEvent`

**Comment by gegy1000** [comment]
A setter shouldn't be needed. You can change the item via event.getItem().setEntityItemStack()

**Comment by iLexiconn** [comment]
Ah, right

**Pull request merged**